src/core/gui/toolbarMenubar/ToolPageLayer.h

```diff
@@ -75,8 +75,8 @@ class ToolPageLayer: public AbstractToolItem, public LayerCtrlListener {
     GtkWidget* layerButton = nullptr;
     GtkWidget* menu = nullptr;
 
-    std::map<int, GtkWidget*> layerItems;
-    std::map<int, GtkWidget*> showLayerItems;
+    std::map<size_t, GtkWidget*> layerItems;
+    std::map<size_t, GtkWidget*> showLayerItems;
```

Comment:
**Question:** What does size_t represent here, an index, Count, hash, some norm, an Apple?

Maybe use something like:
```suggestion
    using AppleJuiceIndex = size_t;
    std::map<AppleJuiceIndex, GtkWidget*> layerItems;
    std::map<AppleJuiceIndex, GtkWidget*> showLayerItems;
```

Also, both std::map and std::unordered_map have huge drawbacks.    
Do holes appear? No + starts with 0? -> std::vector.    
Yes? Must it be sorted? -> leaving it as map might be ok for now (A sorted vector might be better).    
Else -> use std::unordered_map (google::dense_map might be better)

Reply:
Hmm... You're right, there was no reason to have a std::map here to begin with. I replaced it with a std::vector. Good catch!